x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/TokensInvalidationResult.java

```diff
@@ -0,0 +1,141 @@
+ * <ul>
+ * <li>how many of the tokens were actually invalidated</li>
+ * <li>how many tokens are not invalidated in this request because they were already invalidated</li>
+ * <li>how many tokens were not invalidated because of an error and what the error was</li>
+ * </ul>
+ */
+public class TokensInvalidationResult implements ToXContentObject, Writeable {
+
+    private final List<String> invalidatedTokens;
+    private final List<String> previouslyInvalidatedTokens;
+    private final List<ElasticsearchException> errors;
+    private final int attemptCount;
+
+    public TokensInvalidationResult(List<String> invalidatedTokens, List<String> previouslyInvalidatedTokens,
+                                    @Nullable List<ElasticsearchException> errors, int attemptCount) {
+        Objects.requireNonNull(invalidatedTokens, "invalidated_tokens must be provided");
+        this.invalidatedTokens = invalidatedTokens;
+        Objects.requireNonNull(previouslyInvalidatedTokens, "previously_invalidated_tokens must be provided");
+        this.previouslyInvalidatedTokens = previouslyInvalidatedTokens;
+        if (null != errors) {
+            this.errors = errors;
+        } else {
+            this.errors = Collections.emptyList();
+        }
+        this.attemptCount = attemptCount;
+    }
+
+    public TokensInvalidationResult(StreamInput in) throws IOException {
+        this.invalidatedTokens = in.readList(StreamInput::readString);
+        this.previouslyInvalidatedTokens = in.readList(StreamInput::readString);
+        int errorsSize = in.readVInt();
+        List<ElasticsearchException> errors = new ArrayList<>(errorsSize);
+        for (int i = 0; i < errorsSize; i++) {
+            errors.add(in.readException());
+        }
+        this.errors = errors;
+        this.attemptCount = in.readVInt();
+    }
+
+    public static TokensInvalidationResult emptyResult() {
+        return new TokensInvalidationResult(Collections.emptyList(), Collections.emptyList(), Collections.emptyList(), 0);
+    }
+
+    public static TokensInvalidationResult emptyResultWithCounter(int attemptCount) {
+        return new TokensInvalidationResult(Collections.emptyList(), Collections.emptyList(), Collections.emptyList(), attemptCount);
+    }
+
+    public List<String> getInvalidatedTokens() {
+        return invalidatedTokens;
+    }
+
+    public List<String> getPreviouslyInvalidatedTokens() {
+        return previouslyInvalidatedTokens;
+    }
+
+    public List<ElasticsearchException> getErrors() {
+        return errors;
+    }
+
+    public int getAttemptCount() {
+        return attemptCount;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject()
+            .field("invalidated_tokens", invalidatedTokens.size())
+            .field("previously_invalidated_tokens", previouslyInvalidatedTokens.size())
+            .field("error_size", errors.size());
```

Comment:
`error_count` seems more fitting to me.
If we want to know the "size" of the `error_details` array, we can count them.

Reply:
> If we want to know the "size" of the error_details array, we can count them.

I don't doubt that :) but I think it's user friendly to show the count to the user hitting the API instead of requiring them to count the objects in the json array (if they want to know how many errors they got)

Reply:
Sorry, my comment was a bit strange (it was about 11pm when I wrote it)
From an API point of view, we tend to refer to "count" rather than "size" for a countable thing.
See:
> I think it's user friendly to show the count ...

"size" refers to how big something is, not how many there are. It's an array `size`, but an error `.count`

Reply:
Gotcha, I read 

> If we want to know the "size" of the error_details array, we can count them.

without the focus on "size" and "count", and I thought you meant that we don't need the field either way as we can get the same info from the list size. Words... :)